flyteplugins/go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -190,6 +193,31 @@ func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase
 	resource := taskCtx.Resource().(ResourceWrapper)
 	taskInfo := &core.TaskInfo{Logs: resource.LogLinks}
 
+	// If the phase is undefined, we will use state to determine the phase.
+	switch resource.Phase {
+	case flyteIdl.TaskExecution_QUEUED:
+		return core.PhaseInfoQueuedWithTaskInfo(core.DefaultPhaseVersion, resource.Message, taskInfo), nil
+	case flyteIdl.TaskExecution_WAITING_FOR_RESOURCES:
+		return core.PhaseInfoWaitingForResourcesInfo(time.Now(), core.DefaultPhaseVersion, resource.Message, taskInfo), nil
+	case flyteIdl.TaskExecution_INITIALIZING:
+		return core.PhaseInfoInitializing(time.Now(), core.DefaultPhaseVersion, resource.Message, taskInfo), nil
+	case flyteIdl.TaskExecution_RUNNING:
+		return core.PhaseInfoRunning(core.DefaultPhaseVersion, taskInfo), nil
+	case flyteIdl.TaskExecution_SUCCEEDED:
+		err = writeOutput(ctx, taskCtx, resource)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to write output with err %s", err.Error())
+			return core.PhaseInfoUndefined, err
+		}
+		return core.PhaseInfoSuccess(taskInfo), nil
+	case flyteIdl.TaskExecution_ABORTED:
+		return core.PhaseInfoFailure(pluginErrors.TaskFailedWithError, "failed to run the job with aborted phase", taskInfo), nil
+	case flyteIdl.TaskExecution_FAILED:
+		return core.PhaseInfoFailure(pluginErrors.TaskFailedWithError, "failed to run the job", taskInfo), nil
+	}
+
+	logger.Infof(ctx, "Agent returned an undefined phase [%v]. Checking state.", resource.Phase)
```

Comment:
I'm not sure I like this type of fallback logic. Suppose we add a new phase later, the previous switch would not match that and then we fallback to state, which is obviously not what we want.

I prefer explicitly checking whether `resource.Phase` is `UNDEFINED` and then use state.

Reply:
This is a great idea! Thank you so much!

Reply:
Fixed! Thank you again!